### ISSUE TITLE
VS Code: Cody keep showing as Pro even after cancelling subscription from web.

### ISSUE THREAD

**Issue by jay-fibi** [sourcegraph/cody#3108]
### Version

v1.4.0

### Describe the bug

**Steps to reproduce:**

1. Have Cody extension installed in your VS Code
2. User is already logged in to Cody
3. User login to web and navigates to [https://sourcegraph.com/cody/manage](https://sourcegraph.com/cody/manage)
4. Click on "Add Credit Card" button and finish the payment.
5. Click on "Dashboard" and not switched to Cody Pro
6. Open the VS Code editor and check the user is Cody Pro
7. Go to [Subscription](https://accounts.sourcegraph.com/cody/subscription) page and Cancel the subscription
8. Check Cody on the VS Code

**Observed behavior:**
Cody keep showing as Pro tier and able to use Cody Pro features i.e. Choose multiple LLM

### Expected behavior

User should be switched to Free tier on VS Code.

### Additional context

**Video link:**

https://drive.google.com/file/d/1sFb0jUJCAhpm52jcYPieCwU45mF4Yxc_/view?usp=drive_link

**Comment by chrsmith** [comment]
Added to Cody PLG project. Setting as P0 provisionally, since we may want to fix this before 2/21.

I believe this behavior is sort of "by design". Because when you flag your SSC subscription to be canceled at the end of the billing period, the dotcom backend treats this as "being on the pro plan". However, the subscription data does say you are on the Pro Tier, (status==trialing, but with cancelAtPeriodEnd==true).

And so I am guessing that the VS Code client is just seeing status==trialing, and assuming that means "on Cody Pro" and not having the "if trialing AND cancelAtPeriodEnd THEN treat as on Cody Free".

Does that sound right @thenamankumar? I recall you mentioning this behavior, and we kept it as-is because before the SSC launch users could immediately cancel their Cody Pro subscription, and go back to Cody Free. But would we want to change that to avoid this confusing user experience?

In other words, with my understanding of the expected behavior, this is "a bug in the VS Code client" because it isn't aware of this specific _meaning_ of the subscription data... but it seems like we should not require clients reading the subscription data to need to have this type of logic.

**Comment by vdavid** [comment]
Exactly my understanding. The user is still on the Pro plan, just their subscription won't renew.

We don't do refunds, so the fair thing is to let them use what they paid for. Thus, until their subscription period ends, they'll be Pro users.

**Comment by chenkc805** [comment]
Agreed this is expected behavior. Closing
